soc/ite/ec/it51xxx/soc.c

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2025 ITE Corporation. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <soc_common.h>
+#include <zephyr/kernel.h>
+
+static mm_reg_t ecpm_base = DT_REG_ADDR(DT_NODELABEL(ecpm));
+/* it51xxx ECPM registers definition */
+/* 0x03: PLL Control */
+#define ECPM_PLLCTRL 0x03
+
+void chip_pll_ctrl(enum chip_pll_mode mode)
+{
+	volatile uint8_t _pll_ctrl __unused;
+
+	sys_write8(mode, ecpm_base + ECPM_PLLCTRL);
+	/*
+	 * for deep doze / sleep mode
+	 * This load operation will ensure PLL setting is taken into
+	 * control register before wait for interrupt instruction.
+	 */
+	_pll_ctrl = sys_read8(ecpm_base + ECPM_PLLCTRL);
+}
+
+#ifdef CONFIG_SOC_IT51XXX_CPU_IDLE_GATING
+/* Preventing CPU going into idle mode during command queue. */
+static atomic_t cpu_idle_disabled;
+
+void chip_permit_idle(void)
+{
+	atomic_dec(&cpu_idle_disabled);
+}
+
+void chip_block_idle(void)
+{
+	atomic_inc(&cpu_idle_disabled);
+}
+
+bool cpu_idle_not_allowed(void)
+{
+	return !!(atomic_get(&cpu_idle_disabled));
```

Comment:
I think that the `!!` is redundant, seince the return type is a bool the compiler should do this for you

Reply:
Fixed.